### ISSUE TITLE
Enable longdesc / aria-describedby for enhanced web accessibility on images

### ISSUE THREAD

**Issue by Erin-Cecele** [thegetty/quire#852]
### Before proceeding, check to make sure there isn’t an existing issue for this idea or a similar one.

- [X] I have searched the existing issues and determined this is a new proposal.

### Proposed change

The editorial team for GRJ has defined a need to write significantly longer alt texts for some images (particularly the very detailed illustrations.) This has also come up with some of our community members, including @lizneely. 

### Potential Impact

I do not think this will have any major impact on Quire's core code, although I invite @geealbers to chime in if I'm missing anything. 

### Supporting Information

This W3C article lays out the options fairly comprehensively: https://www.w3.org/WAI/tutorials/images/complex/

**Comment by geealbers** [comment]
I'm working on a prototype implementation of this for an upcoming project at Getty. I was originally thinking of putting long description text in an element on the page that was otherwise visually hidden and linking to it from within the `longdesc` attribute on the image. But while I was working, I didn't hide the description right away and I kind of like the result of exposing it for everyone. (Inspired by the National Gallery of Art's work to add image descriptions to their collection pages: https://www.nga.gov/collection/art-object-page.166425.html.) What do people think?

![Screenshot 2024-02-12 at 3 30 52 PM](https://github.com/thegetty/quire/assets/7796401/f01fea0f-e4c1-4043-beb7-ba0b41533530)

The text is in a simple `<details>` html element that has the accessible toggle open/closed built-in. The one thing I'm not sure of is if it's included like this, should we also to add a `longdesc` attribute to the `<img>` element that points to the id of the `<details>` element. Or, is it enough that the details element is labeled as "Image Description" and is contained within the figure caption?

**Comment by cbutcosk** [comment]
@geealbers The way I read https://www.w3.org/WAI/tutorials/images/complex/#approach-3-structurally-associating-the-image-and-its-adjacent-long-description-html5  the `<details>` element should be nested within the `figcaption` and the  `<figure role="group">` handles the rest along with the nested semantics of the control within the caption. Is that possible with the markup you've spec'd above? `aria-describedby` is fine ~but would limit users to text-only long descriptions (eg, links and emphasis might not work as expected)~ (ETA: nevermind, this works ok on Greg's example)

`longdesc` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/img#longdesc) so no need to worry about it.

**Comment by abrin** [comment]
@jasoncorum would you mind taking a quick look and chiming in, I know you know this area a lot better than some of us (myself).

Thank you!

**Comment by geealbers** [comment]
Yes, the prototype approach I outlined above is definitely out of date. I'm interested to hear what light @jasoncorum can shed on the topic. For the sake of the conversation, here's what I've been doing in my most recent implementations, like with [Mummy Portraits, Vol 2](https://github.com/thegetty/mummy-portraits-2):

First, I add the long description directly to the Markdown page under the image (in this example, it's a full HTML table describing a visual graph), classed with `.visually-hidden` (a utility class in Quire that visually hides the element while keeping it accessible to assistive devices) and given an `id`:

```
{% figure 'fig-10-6' %}

<div class="visually-hidden" id="fig-10-6-description">
<table> <thead><tr><th>Angle</th> <th>Number of portraits</th> </tr></thead> <tbody><tr> <td>5, 6</td> <td>&lt;5</td> </tr> <tr> <td>6, 8</td> <td>&lt;5</td> </tr> <tr> <td>8, 10</td> <td>0&lt;</td> </tr> <tr> <td>10, 12</td> <td>&lt;5</td> </tr> <tr> <td>12, 14</td> <td>&lt;5</td> </tr> <tr> <td>14, 16</td> <td>10</td> </tr> <tr> <td>16, 18</td> <td>10&lt;</td> </tr> <tr> <td>18, 20</td> <td>&lt;10</td> </tr> <tr> <td>20, 22</td> <td>&lt;15</td> </tr> <tr> <td>22, 24</td> <td>&lt;20</td> </tr> <tr> <td>24, 26</td> <td>&lt;15</td> </tr> <tr> <td>26, 27</td> <td>15</td> </tr> <tr> <td>27, 29</td> <td>&lt;20</td> </tr> <tr> <td>29, 31</td> <td>&lt;20</td> </tr> <tr> <td>31, 33</td> <td>25&lt;</td> </tr> <tr> <td>33, 35</td> <td>20&lt;</td> </tr> <tr> <td>35, 37</td> <td>5&lt;</td> </tr> <tr> <td>37, 39</td> <td>5&lt;</td> </tr> <tr> <td>39, 41</td> <td>10</td> </tr> <tr> <td>41, 43</td> <td>0&lt;</td> </tr> <tr> <td>43, 45 </td> <td>0&lt;</td> </tr> </tbody></table>
</div>
```

In figures.yaml I reference that `id` with a `described_by_id` attribute:

```
  - id: "fig-10-6"
    src: "figures/fig-10-6.jpg"
    label: "Figure 10.6"
    caption: "Angles generated using distances measured between a nostril and the chin."
    alt: "Bar chart presenting the number of portraits (on the Y axis) that correlate to particular angles (on the X axis) generated using the distance between a nostril and the chin."
    described_by_id: "fig-10-6-description"
```

And then I modified the templates to add that id to the `<img>` tag, ie. `aria-describedby="fig-10-6-description"`.

This was just the simplest way for me to get it done and just a starting point for us here. I won't claim that having users add in HTML elements and ids are a good choice. For me, the important bits (at least the ones I can think of currently) would be:

- Long descriptions need to support HTML markup
- By default they should not be visible on the page 
- But it might be cool to have a config to allow Quire users to instead make the descriptions visible on their site as in my early prototype. In this case though, we'd also need a config to control the text that's in the `<summary>` element ("Image Description" in the example above)
- Long descriptions should definitely appear in the HTML and EPUB output, I'm not sure how they work in PDF?

**Comment by geealbers** [comment]
An additional bonus thought ... the way we ultimately handle these described-by / long description elements for images could also work really well as a way to handle transcripts for audio and video. In both cases, we're talking about a block of HTML that aids in accessibility, would make the figure in question more searchable with pagefind for end users, and would make sense to be exposed visually in a `<details>` element.

**Comment by cbutcosk** [comment]
I tested Greg's spec markup in [Mummy Portraits, vol 2](https://www.getty.edu/publications/mummy-portraits-2/10/), (eg figure 10.6) and VoiceOver + Safari. That setup correctly reads out / and allows interaction with the complex markup of the table pointed to by the `aria-describedby`, with the caveat of an issue around focus trapping when the modal gets activated (#1275). 

When I turned off the screen reader I did miss the unpublished spec markup's details / summary elements for use without assistive devices (part of the WAI guidance also).
